charts/kong/README.md

```diff
@@ -288,6 +293,124 @@ scenario and several settings that are useful when using multiple controllers:
 Standalone controllers require a database-backed Kong instance, as DB-less mode
 requires that a single controller generate a complete Kong configuration.
 
+### Hybrid mode
+
+Kong supports [hybrid mode
+deployments](https://docs.konghq.com/2.0.x/hybrid-mode/) as of Kong 2.0.0 and
+Kong Enterprise 2.1.0. These deployments split Kong nodes into control plane
+(CP) nodes, which provide the admin API and interact with the database, and
+data plane (DP) nodes, which provide the proxy and receive configuration from
+control plane nodes.
+
+The chart does not support deploying control and data plane nodes in a single
+release. Users should [create separate releases for each node
+type](#separate-admin-and-proxy-nodes), i.e. use separate control and data
+plane values.yamls that are installed separately.
+
+* If `.Values.env.role="control_plane"`, the release will create a Service
+  named `RELEASE-NAME-kong-cluster`.
+* Cluster certificates are not yet generated automatically. You must create a
+  certificate/key pair from a local instance, create a Secret containing them,
+  add the Secret to `.Values.secretVolumes`, and add
+  `cluster_cert`/`cluster_cert_key` values indicating its location under
+  `/etc/secrets` to `env`.
+* Note that data plane nodes do not default to using the default port (8005)
+  for `cluster_control_plane`. You should use `RELEASE-NAME-kong-cluster:8005`
+  as the `cluster_control_plane` value if using the default port.
```

Comment:
What I think might be worth documenting is a closed and unambiguous list of steps that the user could blindly follow in order to deploy Kong in hybrid mode. I can see that all the steps (certificate generation, uploading secrets to k8s, updating values yamls, `helm install`) are loosely defined in this doc, but a simple list of steps is something I'd consider useful.

Reply:
Kong and friends are, for better or worse, fairly modular: you can and users do combine different features in surprising and unexpected ways, and there's not necessarily a single path that works for every environment and every configuration combination. This is indeed a source of user confusion, but one that's difficult to avoid given how flexible the product is, so documentation, especially higher-level README stuff, tends towards general rather than verbatim instructions, and assumes some level of familiarity to constrain length and avoid duplication (the latter isn't necessarily a problem on its own, but does turn into one when we need to update docs for new versions across the board--changes are fairly frequent and we do run into issues with doc rot).

The content under `example-values/` is our closest equivalent to this, since while we provide basic step-by-step instructions in the earlier sections of this README, once you get past basics (install Helm, add our repo, create license and Docker Secrets if using Enterprise) the configurations start to diverge quite a bit. Since values.yamls more or less encapsulates everything after, we provide those as templates with instructions for filling in the remaining required items--those were a specific request from sales engineering to fill in the "steps" between initial setup work and release installation.